drivers/mtd/nand/raw/pl35x-nand-controller.c

```diff
@@ -28,7 +28,7 @@
 #include <linux/slab.h>
 #include <linux/clk.h>
 
-#define PL35X_NANDC_DRIVER_NAME "pl35x-nand-controller"
+#define PL35X_NANDC_DRIVER_NAME "xilinx_nand"
```

Comment:
Cherry picked one more commit (https://github.com/ni/linux/commit/df465252df84668ac5318fe6394eca5e3fe38426) as we're observing the same issue with driver name not lining up with what u-boot provides as boot arguments for partitioning the nand.

Reply:
Gah, this is coming back to haunt me. I really wish we would have fixed it in u-boot at the time. I guess we don't have a choice now and have to keep this around.